pkg/apis/broker/v1beta1/broker_validation.go

```diff
@@ -43,6 +43,9 @@ func ValidateDeliverySpec(ctx context.Context, spec *eventingduckv1beta1.Deliver
 	if spec.BackoffPolicy == nil {
 		errs = errs.Also(apis.ErrMissingField("backoffPolicy"))
 	}
+	if spec.Retry != nil && spec.DeadLetterSink == nil {
+		errs = errs.Also(apis.ErrMissingField("deadLetterSink"))
```

Comment:
I would prefer an error message saying that if `spec.retry` is set, then `spec.deadLetterSink` must be set. 

As-is, I think this may confuse people into believing that `spec.deadLetterSink` is always required.

Reply:
Make sense. Updated the error to: "need DeadLetterSink when retry is defined: spec.delivery.deadLetterSink"